packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/SplitButtonWidget.tsx

```diff
@@ -0,0 +1,352 @@
+    }
+  }, [selection, widget]);
+
+  const classes = useStyles();
+
+  const handleMenuItemClick = (_event, index) => {
+    setState((prevState) => ({ ...prevState, open: false, selectedIndex: index }));
+  };
+
+  const handleToggle = () => {
+    setState((prevState) => ({ ...prevState, open: !prevState.open }));
+  };
+
+  const handleClose = (_event) => {
+    if (ref.current && ref.current.contains(event.target)) {
+      return;
+    }
+    setState((prevState) => ({ ...prevState, open: false }));
+  };
+
+  const handleDragEnter: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.add(classes.dragOver);
+  };
+
+  const handleDragOver: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.add(classes.dragOver);
+  };
+
+  const handleDragLeave: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.remove(classes.dragOver);
+  };
+
+  const handleDrop: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.remove(classes.dragOver);
+
+    const id: string = event.dataTransfer.getData('draggedElementId');
+    const type: string = event.dataTransfer.getData('draggedElementType');
+    if (type !== 'Widget') {
+      return;
+    }
+
+    if (id === 'Button') {
+      const addWidgetInput: GQLAddWidgetInput = {
+        id: crypto.randomUUID(),
+        editingContextId,
+        representationId,
+        containerId: widget.id,
+        kind: id,
+        index: widget.actions.length || 0,
+      };
+      const addWidgetVariables: GQLAddWidgetMutationVariables = { input: addWidgetInput };
+      addWidget({ variables: addWidgetVariables });
+    }
+  };
+
+  const getStyle = (index: number): React.CSSProperties => {
+    if (widget.actions.length == 0) {
+      const defaultProps = {
+        backgroundColor: null,
+        foregroundColor: null,
+        fontSize: null,
+        italic: null,
+        bold: null,
+        underline: null,
+        strikeThrough: null,
+      };
+      return actionStyle(theme, defaultProps);
+    }
+    return actionStyle(theme, widget.actions[index].style);
+  };
+
+  const getIconSrc = (originalSrc: string): string => {
+    if (originalSrc.includes('aql:')) {
+      const index = originalSrc.indexOf('aql:');
+      const newPrefix = originalSrc.slice(0, index);
+      const removedPrefix = originalSrc.slice(index + 5);
+      const removedSuffix = removedPrefix.slice(0, removedPrefix.length - 1);
+      return newPrefix + removedSuffix;
+    }
+    return originalSrc;
+  };
+
+  return (
+    <div className={classes.containerAndLabel} tabIndex={0} ref={ref}>
+      <div className={classes.propertySectionLabel}>
+        <Typography variant="subtitle2" className={state.selected ? classes.selected : ''}>
+          {widget?.label || 'Lorem'}
```

Comment:
Are you sure that it should be `Lorem` here? For the title of a widget, If no name/label has been set by the user, we use the name of the widget class. Please verify and fix it if needed.